box: rollback txns in WAL queue before in-flight tnxs (#11279)
tarantool/tarantool: changelogs/unreleased/gh-11179-wal-queue-rollback.md

## bugfix/box

> * Fixed a bug when txns in WAL queue are not rollbacked on WAL error (gh-11179).

Reviewer: ```suggestion
* Fixed a bug when transactions in the WAL queue were not rolled back on WAL error (gh-11179).
```

Author: Fixed.
